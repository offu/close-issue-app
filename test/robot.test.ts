import { createRobot } from 'probot'
import { IssuesCreateCommentParams, IssuesEditParams, ReposGetContentParams } from '@octokit/rest'
import * as path from 'path'
import * as fs from 'fs'
import app = require('../src/robot')
import payload from './fixtures/issues.event.json'

describe('robot', () => {
  let robot
  let github
  const exampleConfig = fs.readFileSync(path.resolve(__dirname, '../example.config.yml'), 'utf-8')

  beforeEach(() => {
    robot = createRobot(null)
    app(robot)
    github = {
      issues: {
        createComment: jest.fn().mockResolvedValue(null),
        edit: jest.fn().mockResolvedValue(null)
      },
      repos: {
        getContent: jest.fn().mockResolvedValue({ data: {
          content: exampleConfig,
          encoding: 'utf-8'
        }})
      }
    }
    robot.auth = () => Promise.resolve(github)
  })

  describe('robot deal with issues', () => {
    it('invalid issue', async () => {
      await robot.receive(payload)
      const params = {
        owner: 'baxterthehacker',
        repo: 'public-repo',
        number: 2
      }
      const createCommentParams: IssuesCreateCommentParams = { ...params, body: 'test' }
      const editIssueParams: IssuesEditParams = { ...params, state: 'closed' }
      const getContentParams: ReposGetContentParams = {
        owner: 'baxterthehacker',
        repo: 'public-repo',
        path: '/.github/issue-close-app.yml'
      }
      expect(github.issues.createComment).toHaveBeenCalledWith(createCommentParams)
      expect(github.issues.edit).toHaveBeenCalledWith(editIssueParams)
      expect(github.repos.getContent).toHaveBeenCalledWith(getContentParams)
    })
  })
})
