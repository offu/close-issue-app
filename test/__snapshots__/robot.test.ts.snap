// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`robot robot config without default config 1`] = `[MockFunction]`;

exports[`robot robot config without default config 2`] = `[MockFunction]`;

exports[`robot robot config without default config 3`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "owner": "baxterthehacker",
        "path": "/.github/issue-close-app.yml",
        "repo": "public-repo",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Promise {},
    },
  ],
}
`;

exports[`robot robot deal with issues get error 1`] = `
Object {
  "number": 2,
  "owner": "baxterthehacker",
  "repo": "public-repo",
}
`;

exports[`robot robot deal with issues invalid issue 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "body": "test",
        "number": 2,
        "owner": "baxterthehacker",
        "repo": "public-repo",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Promise {},
    },
  ],
}
`;

exports[`robot robot deal with issues invalid issue 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "number": 2,
        "owner": "baxterthehacker",
        "repo": "public-repo",
        "state": "closed",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Promise {},
    },
  ],
}
`;

exports[`robot robot deal with issues invalid issue 3`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "owner": "baxterthehacker",
        "path": "/.github/issue-close-app.yml",
        "repo": "public-repo",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Promise {},
    },
  ],
}
`;
